### PR TITLE
fix rng_state

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
When loading from saved ckpt, the rnd_state should be byte type.

### CODE REVIEW

File: alf/optimizers/optimizers.py [hunk 1]
```diff
@@ -252,7 +252,8 @@ def _adjust_capacity(self, capacity_ratio: float, old_param_values: Dict):
                             'rng_state'] = self._random_number_generator.get_state(
                             )
                     else:
-                        rng_state = state['rng_state']
+                        # the state['rng_state'] could be float tensor if loaded from ckpt
+                        rng_state = state['rng_state'].cpu().byte()
```
Comment (emailweixu) [inline alf/optimizers/optimizers.py]: This should be done in load_state_dict.

Comment (Haichao-Zhang) [reply alf/optimizers/optimizers.py]: Done

File: alf/optimizers/optimizers.py [hunk 2]
```diff
@@ -482,6 +482,22 @@ def add_param_group(self, param_group):
...
+        This function first call the parent's ``load_state_dict()`` function, and
+        then make sure the ``rng_state`` is in correct data type.
+        """
+        super(NewCls, self).load_state_dict(state_dict)
+        if len(self.state) > 0:
+            for param_group in self.param_groups:
+                for p in param_group['params']:
+                    state = self.state[p]
+                    if 'rng_state' in state:
+                        state['rng_state'] = state['rng_state'].cpu().byte()
```
Comment (emailweixu) [inline alf/optimizers/optimizers.py]: Is rng_state always in cpu?

Comment (Haichao-Zhang) [reply alf/optimizers/optimizers.py]: Handled it in a more general way by using ``_rng_state_device``

**Review by emailweixu — APPROVED** [review]

**Pull request merged**